src/OSPSuite.Core/Domain/Services/ModelValidator.cs

```diff
@@ -66,22 +67,26 @@ protected void CheckFormulaIn(IUsingFormula entity, IFormula formulaToCheck, Res
          var entityAbsolutePath = _objectPathFactory.CreateAbsoluteObjectPath(entity).ToPathString();
          var builder = _simulationBuilder.BuilderFor(entity);
          var objectWithError = builder ?? entity;
+         void checkPathInEntity(ObjectPath objectPath) => CheckPath(entity, objectPath, resolveErrorBehavior);
 
-         // Dynamic formula may contain object path that will be resolved per instance. It cannot be checked here
-         if (formulaToCheck.IsDynamic())
-            return;
-
-
-         if (formulaToCheck.IsBlackBox())
+         switch (formulaToCheck)
          {
-            addNotificationType(NotificationType.Error, objectWithError, Validation.FormulaIsBlackBoxIn(entity.Name, entityAbsolutePath));
-            return;
-         }
+            // Dynamic formula may contain object path that will be resolved per instance. It cannot be checked here
+            case DynamicFormula _:
+               return;
+            case BlackBoxFormula _:
+               addNotificationType(NotificationType.Error, objectWithError, Validation.FormulaIsBlackBoxIn(entity.Name, entityAbsolutePath));
+               return;
 
-         foreach (var objectPath in formulaToCheck.ObjectPaths)
-         {
-            CheckPath(entity, objectPath, resolveErrorBehavior);
+            //Table formula with offset may use an x argument that is not defined. A dynamic check will be done in the formula itself
+            //so in this case, we only check the path to the table that should be defined
+            case TableFormulaWithXArgument formulaWithXArgument:
+               checkPathInEntity(formulaWithXArgument.FormulaUsablePathBy(formulaWithXArgument.TableObjectAlias));
```

Comment:
Also here, during the validation, we will not check all object path. Rather only the Table will be checked